cookbook/agents/agent_run_cancellation.py

```diff
@@ -0,0 +1,76 @@
+"""🛑 Agent Run Cancellation - Stopping Long-Running Tasks
+
+This example shows how to cancel agent runs mid-execution.
+Perfect for scenarios where users want to stop long-running tasks like:
+- Complex research or analysis 
+- Large document processing
+- Time-consuming calculations
+
+Run `pip install openai agno` to install dependencies.
+"""
+
+import asyncio
+
+from agno.agent import Agent
+from agno.models.openai import OpenAIChat
+
+# Create a research agent that might take a while
+research_agent = Agent(
+    model=OpenAIChat(id="gpt-4o-mini"),
+    description="Research specialist who performs in-depth analysis",
+    instructions=[
+        "Provide comprehensive, detailed analysis",
+        "Include multiple perspectives and examples",
+        "Take your time to be thorough"
+    ],
+    markdown=True,
+)
+
+async def demonstrate_cancellation():
+    """Show how to cancel a long-running agent task"""
+    
+    print("🚀 Starting a comprehensive research task...")
+    
+    # Start a potentially long-running task
```

Comment:
I think make clear you are starting it in a thread, to demonstrate it running in a separate thread